File: Top-Interview-Questions/Math-Easy-Collection/412-fizz-buzz.js
/**
 * @param {number} n
 * @return {string[]}
 * Runtime: 72 ms, faster than 53.34% of JavaScript online submissions for Fizz Buzz.
 * Memory Usage: 37.6 MB, less than 50.85% of JavaScript online submissions for Fizz Buzz.
 * TC: O(n)
 * SC: O(1), if we exclude returning memory
 */
var fizzBuzz = function(n) {
    const result = [];

    for (let i = 1; i <= n; i++) {
        const fizz = i % 3 === 0 ? "Fizz" : "";
        const buzz = i % 5 === 0 ? "Buzz" : "";

        if (fizz.length === 0 && buzz.length === 0) {
            result.push(i.toString());
        } else {
            result.push(`${fizz}${buzz}`);
        }
    }

    return result;
};

File: Top-Interview-Questions/Math-Easy-Collection/412-Fizz-Buzz.cpp
// TC: O(n)
// SC: O(n)
// Runtime: 8 ms, faster than 55.81%
// Memory Usage: 7.9 MB, less than 58.19% 
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> result;
        
        for(int i = 1; i <= n ; i++){
            
            if (!(i%3) && !(i%5))
               result.push_back("FizzBuzz");
            else if (!(i%5))
               result.push_back("Buzz");
            else if (!(i%3))
               result.push_back("Fizz");
            else
               result.push_back(to_string(i));
        }
        
        return result;
    }
};